packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts

```diff
@@ -519,11 +574,83 @@ export class EndpointAccess {
   }
 }
 
+/**
+ * Options for configuring EKS Auto Mode compute settings.
+ * When enabled, EKS will automatically manage compute resources like node groups and Fargate profiles.
+ */
+export interface ComputeConfig {
+  /**
+   * Names of nodePools to include in Auto Mode.
+   * You cannot modify the built in system and general-purpose node pools. You can only enable or disable them.
+   * Node pool values are case-sensitive and must be `general-purpose` and/or `system`.
+   *
+   * @see - https://docs.aws.amazon.com/eks/latest/userguide/create-node-pool.html
+   *
+   * @default - ['system', 'general-purpose']
+   */
+  readonly nodePools?: string[];
```

Comment:
At this moment, EKS only allow you to opt in `system` and/or `general-purpose`. For future-proofing, I am making it a string[] with validations rather then a enum. This allows we to have the freedom to support more types in the future without BCs.